src/nvim/option.c

```diff
@@ -3558,24 +3562,22 @@ static const char *did_set_option(int opt_idx, void *varp, OptVal old_value, Opt
     .os_win = curwin
   };
 
-  if ((int *)varp == &p_force_on) {
-    did_set_cb = did_set_force_on;
-  } else if ((int *)varp == &p_force_off) {
-    did_set_cb = did_set_force_off;
-  } else {
-    did_set_cb = opt->opt_did_set_cb;
+  // Disallow changing immutable options
```

Comment:
> Why not simply disallow _setting_ an immutable option?

Because the old behavior is to allow setting it if the value is the same as before.

Reply:
> Why not simply disallow _setting_ an immutable option?

Because the old behavior is to allow (silent ignore) setting it if the value is the same as before. If breaking backwards compatibility is fine, then I have no problems with simply not allowing the value to be set